ckanext/recombinant/views.py

```diff
@@ -440,18 +440,30 @@ def preview_table(resource_name, owner_org, errors=None):
     except RecombinantException:
         abort(404, _('Recombinant resource_name not found'))
 
-    if 'create' in request.form:
+    if 'create' in request.form or 'refresh' in request.form:
         # check if the user can update datasets for organization
         # admin and editors should be able to init recombinant records
         if not has_user_permission_for_group_or_org(org_object.id, g.user, 'update_dataset'):
             abort(403, _('User %s not authorized to create packages') % (str(g.user)))
         try:
+            # check if the dataset exists
             dataset = lc.action.recombinant_show(
                 dataset_type=chromo['dataset_type'], owner_org=owner_org)
+            # check that the resource has datastore tables
+            for _r in dataset['resources']:
+                if _r['name'] == resource_name:
+                    lc.action.datastore_search(
+                        resource_id=_r['id'], limit=0)
+                    break
```

Comment:
how about moving the check for the status of recombinant resources into recombinant_show? Seems like it would be useful in places other than this view. Also is it enough to check the datastore_active flag?

Reply:
Ah, apparently the `recombinant_show` already does this:
```
try:
            ds = lc.action.datastore_search(
                resource_id=resource['id'],
                limit=1)
            datastore_correct = _datastore_match(r['fields'], ds['fields'])
            out['datastore_correct'] = datastore_correct
            resources_correct = resources_correct and datastore_correct
            out['datastore_rows'] = ds.get('total', 0)
        except NotFound:
            out['error'] = 'datastore table missing'
            resources_correct = False
```

So I just am now using `if error in resource` throughout the templates now.

Reply:
but why do we still need to loop through resources and datastore_search here in the view?

Reply:
@wardi oops forgot to change it in the above try/catch. Okay, it is not just checking if the recombinant resource has `error`, then raises.